aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -89,6 +91,34 @@ impl TransactionMetadata {
                 TransactionPayload::Multisig(m) => Some(m.clone()),
                 _ => None,
             },
+            raw_txn_hash: match txn.authenticator_ref() {
+                MultiAgent {
+                    sender,
+                    secondary_signer_addresses,
+                    secondary_signers,
+                } => {
+                    let raw_txn = RawTransactionWithData::new_multi_agent(
+                        txn.clone().into_raw_transaction(),
+                        secondary_signer_addresses.clone(),
+                    );
+                    raw_txn.hash().to_vec()
+                },
+                FeePayer {
+                    sender,
+                    secondary_signer_addresses,
+                    secondary_signers,
+                    fee_payer_address,
+                    fee_payer_signer,
+                } => {
+                    let raw_txn = RawTransactionWithData::new_fee_payer(
+                        txn.clone().into_raw_transaction(),
+                        secondary_signer_addresses.clone(),
+                        *fee_payer_address,
```

Comment:
Let's make this ZERO instead...
We don't want to have two possible valid hash here.
cc @davidiw

Reply:
We could make the function return a vector of hash values to return both: `[hash_for_0x0_fee_payer, hash_for_actual_fee_payer]`.

Reply:
what if it is not a fee payer txn? which two hashes would you return?
Also, I think users will be super confused.

Reply:
> what if it is not a fee payer txn? which two hashes would you return?
> Also, I think users will be super confused.

If it's not a fee-payer transaction, we could return a singleton vector with the hash of the raw transaction `[hash_value].`

Reply:
So it can be either 1 or 2? 
Don't you think it is too confusing?

Reply:
> So it can be either 1 or 2?

Yes, 2 return values for fee-payer transactions and 1 return value for other types of transactions.

> Don't you think it is too confusing?

I think the more important question is whether it's necessary. The current authentication process utilizes both hash values with the actual fee payer and without it. Would there be any scenario where an AA user needs both hash values?

Reply:
Each txn must be either case but not both. Do you have any data on this, like how many people are using the each signing pre-image? We will note in the function that the hash is only for all zero if necessary.

I don't think there is a case that they need both. Do you have a use case in mind?

Reply:
The fee payer must sign the transaction with their own address included. The primary and secondary signers have two options: (1) Signing the transaction with the fee payer's address included, or (2) Signing the transaction with 0x0 (the zero address) as the fee payer address. I believe that option (2) is commonly used when the specific fee-payer account is determined later in the process. However, I don't have specific data to confirm this. Perhaps @davidiw, @gregnazario, or @movekevin may have more context on this.

Reply:
The second path is important.

The sender initially signs with 0x0, and then the fee payer signs with their address in that slot.

There are then 2 hashes, it would be the fee payer's responsibility to tell the user what the new hash is.

Though, this definitely points out a use case for a wait on sequence number function

Reply:
@lightmark, per request, I've updated the function to generate the hash value using 0x0 as the fee-payer address. If we need to generate hash values based on actual fee-payer addresses, we will need to introduce a separate API, as you suggested.